ext/node/polyfills/internal/crypto/util.ts

```diff
@@ -46,6 +46,52 @@ const digestAlgorithms = [
   "sha1",
 ];
 
+export type EllipticCurve = {
+  name: string;
+  ephemeral: boolean;
+  privateKeySize: number;
+  publicKeySize: number;
+  sharedSecretSize: number;
+};
+
+export const ellipticCurves: Array<EllipticCurve> = [
+  {
+    name: "secp256k1",
+    ephemeral: false,
+    privateKeySize: 32,
+    publicKeySize: 65,
+    sharedSecretSize: 32,
+  }, // Weierstrass-class EC used by Bitcoin
+  {
+    name: "prime256v1",
+    ephemeral: true,
+    privateKeySize: 32,
+    publicKeySize: 65,
+    sharedSecretSize: 32,
+  }, // NIST P-256 EC
+  {
+    name: "secp256r1",
+    ephemeral: true,
+    privateKeySize: 32,
+    publicKeySize: 65,
+    sharedSecretSize: 32,
+  }, // NIST P-256 EC (same as above)
+  {
+    name: "secp384r1",
+    ephemeral: true,
+    privateKeySize: 48,
+    publicKeySize: 97,
+    sharedSecretSize: 48,
+  }, // NIST P-384 EC
+  {
+    name: "secp224r1",
+    ephemeral: true,
+    privateKeySize: 28,
+    publicKeySize: 57,
+    sharedSecretSize: 28,
+  }, // NIST P-224 EC
+];
```

Comment:
Perhaps `as const satisfies Array<EllipticCurve>` would be better.

Reply:
Why?

Reply:
Because the type becomes clearer. Please see [this TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAogNnAlmYiDGBhArgJwG7QC8UA3gFBRQB2AhgLYQBcUAzsDolQOYDcFUEMAAsIDHDTjMARgHsZcCDSp9KYDnhrAIAaQggAyogBeTaljpSIOFVDBYpSNLoPHTVc5ev8WQmjggAJvoQaP7AhibM7hZWfAC+fGRoMlRsUGi4-lTAzPBIKOjY+BAA2gC6UMQl-AD0NVAAegD8-OSUlLQMzABELCFgAEwArABsANYAjN0ANPyUgiJiEswAZhJ9s+226po6ehGmAMwDm+12DujOB8wjQ6eUPn6BwaEQ4a7Mx6dx01B1UAB1CCIKxscQsFgAWjQcBoENgGCgWD6ASgUhAUAAQohgMlOK05tR6KZumpEAxhiM8FN7gJhKIrMsoOwsBBaWSNForh8oF9CedHNzIlBbrTHv4giEwtdeSd+D8-vUAHIASX0ABUoAAFSGUhEEradEl9NCDUY4GmEhYM8SSZk4VnsnZc-Y8vlbAWXV3C0WE8XPKVvGXuqAK-6qjXa3WjBFQAAULGJUDhydkBAAlAb2kaeiawIcABwAFgtMyt9KWdpZbP5zr2LmFRYL7Psgu9pgAnAB2MW+CUvaU8pvfX7htWanWFov6yhtbPE3P9AYDEuWrbWyvMatOxCc+sygbN-mtr0N0xDHt+vsB17vYWHkeKqARie6lf6sqJZKpYCsLBcLgIDSKpanqZosw6BcoF6JdRkmMt1wrRk7TWOANlrXddiFI45Q9E8nHbG47ivJ5JVvYNcNDUd6iBEEcDBOEoRhRjY2RQI0QxbFcRkfFZ0JHNoLJClRmpBD2g3ZCtwdGsPTrbDPko1R8PkkViK2f0yMHYUQzDZVxyjPUYAwCCiS6aC80pUtaQk20pMdDC9xUkMlIuAizyI3tSIHIM3Uo3Tn30nVDMRBMkxTGg0wgTM+MNKCYNNKcrPLRZJPtezZMwl13KgYdj1clTu08-tAzvUxcsofyXyjKcZ1Ifi4oslcksQlLbLSmSzjkwioAfPK22yi8ipvLTTF6irqICyMgvfIyyAqFNvzSRNUBYFYQRYKB-BoAIUjgDE8mQVBMFwAhyj4IA).